src/Common/src/TypeSystem/Common/InstantiatedType.cs

```diff
@@ -281,5 +281,21 @@ public override bool HasCustomAttribute(string attributeNamespace, string attrib
         {
             return _typeDef.HasCustomAttribute(attributeNamespace, attributeName);
         }
+
+        public override MetadataType GetNestedType(string name)
+        {
+            // It doesn't make sense to look at the types nested in an instantiated type in general.
+            // In particular, if the outer type is generic, and the inner type has an unrelated set of 
+            // generic parameters, the result is certain to be confusing.
+            throw new NotImplementedException();
```

Comment:
The only people that would be weirded out by having an implementation that returns open nested types would be people who got exposed to the weird and crazy world of [CCI](http://ccimetadata.codeplex.com/). We already differ from CCI in our handling of nested generic types anyway, so it's probably not a big deal to differ in here too.

I would say let's go with @manu-silicon suggestion on this.